src/Microsoft.Health.Fhir.Shared.Api/Controllers/FhirController.cs

```diff
@@ -389,6 +390,22 @@ public async Task<IActionResult> Delete(string typeParameter, string idParameter
             return FhirResult.NoContent().SetETagHeader(response.WeakETag);
         }
 
+        /// <summary>
+        /// Patches the specified resource
+        /// </summary>
+        /// <param name="typeParameter">The type.</param>
+        /// <param name="idParameter">The identifier.</param>
+        [HttpPatch]
+        [Route(KnownRoutes.ResourceTypeById)]
+        [AuditEventType(AuditEventSubType.Patch)]
+        [Authorize(PolicyNames.WritePolicy)]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Performance", "CA1822:Mark members as static", Justification = "Controller methods won't be called if static.")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "CA1801:Review unused parameters", Justification = "Need the parameters for routing to work.")]
+        public Task<IActionResult> Patch(string typeParameter, string idParameter)
+        {
+            throw new MethodNotAllowedException(Resources.PatchNotSupported);
```

Comment:
nit: maybe we can re-use `Resources.UnsupportedOperation` here instead of creating a new one?

Reply:
Patch is an http method, not an [FHIR operation](https://www.hl7.org/fhir/operations.html).